Automattic/woocommerce-payments
# Tidying up and improving the React logic for Stripe components

Fixes #9409 

#### Changes proposed in this Pull Request





* Moving API calls to the `utils` file to simplify the logic
* Removing some unneeded code and tidying up the error handling.
* Some updates to the type definitions for consistency
* Some updates to how we handle sessions







#### Testing instructions





* Remove any existing account you have on your store (or using a JN site).
* In the dev tools, ensure `Embedded KYC` is enabled in the feature selection.
* Open the network tab on your dev tools in your browser.
* Start an onboarding. When you get to the Embedded KYC step, look for the request to the `/onboarding/kyc/session` endpoint. Check the value of `expiresAt` and make a note of it.
* Wait until _after_ the `expiresAt` expires before proceeding with the embedded KYC onboarding.
* You should not see an error about the session being expired. The onboarding should work as expected.
* Also try leaving the onboarding midway through embedded KYC and going back to it (via the Connect page) - should work as expected.
* After the onboarding, you should be redirected to the Overview page.
* Do another onboarding (from the start), this time with a Progressive Account and ensure that the progressive onboarding also works as expected. 

-------------------


**Post merge**
